crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -500,9 +500,9 @@ where
         hash: B256,
     ) -> RpcResult<Option<RpcTransaction<T::NetworkTypes>>> {
         trace!(target: "rpc::eth", ?hash, "Serving eth_getTransactionByHash");
-        Ok(EthTransactions::transaction_by_hash(self, hash)
-            .await?
-            .map(|tx| tx.into_transaction(self.tx_resp_builder())))
+        Ok(EthTransactions::transaction_by_hash(self, hash).await?.map(|tx| {
+            tx.into_transaction(self.tx_resp_builder()).expect("failed to fill transaction")
```

Comment:
checkout the `FromEthApiError` trait and how its done in other impls of the api e.g. `.map_err(Self::Error::from_eth_err)?`

Reply:
I see what you mean, then the easiest way is just to use the same error type as `EthApiTypes::Error`, i.e. `OpEthApiError` for optimism and `EthApiError` for eth, will push a commit to help you along